docs/hosting.md

```diff
@@ -161,3 +161,123 @@ kubectl get service
+
+- On EC2 Dashboard, click on **“Launch Instance”.**
+
+- Choose an AMI (Amazon Machine Instance). Here choose, **Amazon Linux AMI 2018.03.0** as it comes under Free Tier.
+
+- Choose an Instance Type. Here, choose **t2.micro**. You can opt for a bigger machine to deal with a powerful model or are willing to pay.
+
+- Configure Instance Details, Add Storage and Tags (_All default value, no change, click on Next_).
+
+- Configure Security Group *
+
+    1. SSH (port 22) is by default added to Inbound Rule.
+
+    2. ** Click on Add Rule Type = Custom TCP Rule, Port Range = 8501 and Source: Anywhere.
+    ** _We use the port 8501 here since it is the custom port used by Streamlit._
+
+    3. Ignore Warning as of now
+
+
+- Click on Review & Launch
+
+    1. _Review the configuration you provided, ignore the warning about Security Group as of now._
+    2. Click on **“Launch”**
+
+- A pop-up will appear and ask you to Create a new key-pair (Public-Private)
+
+    1. Select - Create a new key pair
+    2. Provide key pair name = img_ai_app_streamlit
+    3. Click on “Download Key Pair”
+
+        :white_check_mark: Keep this key safe as it would be required every time you need to login to this particular machine.
+
+    4. Click on **“Launch Instance”** after downloading the key pair
+
+- Click on **View Instances.** You can now go to your instances to see if your instance has started.
+
+### EC2 Setup form Local Machine
+
+*There are several methods of getting access into our instance using SSH. If you are on a Unix based system such as Linux or macOs you will have SSH already. For Windows users you will need to use putty and puttygen to gain access.*
+
+**For *Unix* users**
+
+1. Your key must not be publicly viewable for SSH to work. Use this command:
+
+``` console
+chmod 400 img_ai_app_streamlit.pem
+```
+2. Then, connect to your instance using its Public DNS:
+
+``` console
+ssh i img_ai_app_streamlit.pem ec2-user@your_public_dns_address
+```
+
+Select your instance, and copy the Public DNS(IPv4) Address from the description. It should be something starting with ec2.
+
+**For *Windows* users**
+
+- Follow [this]( https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/putty.html?icmpid=docs_ec2_console).
+
+Now we are in our AWS EC2 Instance.
+
+#### Setting Up Environment
+
+- The next step is to install python and all the necessary libraries for your app.
+
+```console
+sudo yum install python3 git
+```
+
+- After these two installation we can use git to get our app within our instance.<br>
+If you are on **Windows** you can also use **WinSCP** to copy your files to your instance. Refer [this]( https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/putty.html#Transfer_WinSCP).
+
+- Now, clone your repository into instance.
+
+```console
+ git clone link_to_your_repository.git
+```
+
+- The next step is to  `cd `  into repository and install required packages using
+
+```console
+python3 -m pip install streamlit
```

Comment:
These are to be done from requirements.txt or using the docker container

Reply:
Sorry sir, I did not know about docker.